pkg/cli/admin/inspect/pod.go

```diff
@@ -88,14 +89,25 @@ func filterContainerLogsErrors(err error) error {
 
 func (o *InspectOptions) gatherContainerRotatedLogFiles(destDir string, pod *corev1.Pod, container *corev1.Container) error {
 
+	var rotatedLogFilename string
+	if value, exists := pod.Annotations["kubernetes.io/config.source"]; exists && value == "file" {
```

Comment:
Using URL source for pulling static pods is probably not supported for the control plane since it poses a security risk. All KA, KCM and KS are storing the manifests as a file. I also assume we do not support customer extensions of the control plane. So there's a high chance the http source will never get used in OpenShift.

@soltysh I wonder what is your though on that?

Reply:
@ingvagabund  One thing: There may be other use cases for static pods other than control plane extensions. I guess that whatever needs to run in a node before the kube-apiserver is reachable via network might be a candidate for a static pod.

Honestly, it would be good to define the exact support status of static pods (whether it is supported that an OpenShift user creates his/her own static pods), including whether it would be supported to pass the `--manifest-url` required to use URL-based static pods. However, I am not the one who can make such a decision (I am just from support).

As the main point of impact is in the static pods used by the control plane and those are file-based only (because they are rolled out by their respective cluster operators as files), wouldn't it make sense to focus on the file case for now and defer the URL thingy for a later bug fix if needed?

Reply:
> it would be good to define the exact support status of static pods

That's up to a broader discussion. Added the topic to our agenda for discussion.

> As the main point of impact is in the static pods used by the control plane and those are file-based only, wouldn't it make sense to focus on the file case for now and defer the URL thingy for a later bug fix if needed?

@ardaguclu I prefer to go this way so in case this options is missing we get an RFE from a customer. Through which we can learn someone is actually deploying static pods through the http.

Reply:
Sounds good @ingvagabund

Reply:
> Honestly, it would be good to define the exact support status of static pods.

@palonsoro conclusion from today's discussion is there's no official restrictions preventing anyone creating any additional static pod using any existing delivery method.